homeassistant/components/zha.py

```diff
@@ -0,0 +1,293 @@
+"""
+Support for ZigBee Home Automation devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/zha/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant import const
+from homeassistant.helpers import discovery, entity
+from homeassistant.util import slugify
+import homeassistant.helpers.config_validation as cv
+
+
+# Definitions for interfacing with the rest of HA
+REQUIREMENTS = ['bellows==0.2.2']
+
+DOMAIN = 'zha'
+
+CONF_USB_PATH = 'usb_path'
+CONF_DATABASE = 'database_path'
+CONF_DEVICE_CONFIG = 'device_config'
+DATA_DEVICE_CONFIG = 'zha_device_config'
+
+DEVICE_CONFIG_SCHEMA_ENTRY = vol.Schema({
+    vol.Optional(const.CONF_TYPE): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        CONF_USB_PATH: cv.string,
+        CONF_DATABASE: cv.string,
+        vol.Optional(CONF_DEVICE_CONFIG):
+            vol.Schema({cv.string: DEVICE_CONFIG_SCHEMA_ENTRY}),
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+ATTR_DURATION = "duration"
+
+SERVICE_PERMIT = "permit"
+SERVICE_DESCRIPTIONS = {
+    SERVICE_PERMIT: {
+        "description": "Allow nodes to join the Zigbee network",
+        "fields": {
+            "duration": {
+                "description": "Time to permit joins, in seconds",
+                "example": "60",
+            },
+        },
+    },
+}
+SERVICE_SCHEMAS = {
+    SERVICE_PERMIT: vol.Schema({
+        vol.Optional(ATTR_DURATION, default=60):
+            vol.All(vol.Coerce(int), vol.Range(1, 254)),
+    }),
+}
+
+
+# ZigBee definitions
+CENTICELSIUS = 'C-100'
+
+# Device types which require more than one cluster to work together
+DEVICE_TYPES = {
+    260: {  # ZHA
```

Comment:
> unless you're also planning on further expanding bellows to support a variety of dissimilar hardware.

That's how all home automation software is build. Someone will miss a piece and will decide to contribute it. Just like this PR does 👍